### ISSUE TITLE
Authentication failed since 1.0.54 (Private Satis with Basic-Auth )

### ISSUE THREAD

**Issue by alexander-bogomolov** [symfony/flex#273]
Since the 1.0.54 release the install/requere command ask for authentication. We use a private Satis instance secured by Basic-Auth.
![screen-001](https://user-images.githubusercontent.com/127733/34666612-7faea520-f465-11e7-82dc-9f7a232977d5.png)

The data ist stored in the auth.json an works well with Flex < 1.0.53. Entering the username and password in the CLI fails also.

I think the symfony/thanks component try to connect the Satis.

**Comment by nicolas-grekas** [comment]
> I think the symfony/thanks component try to connect the Satis.

`symfony/thanks` doesn't do any request unless you explicitly call `composer thanks`

The issue must be related to #261instead.

**Comment by biomedia-thomas** [comment]
We also have some issues with the prefetcher in combination with a custom satis repos that requires authentication. Maybe this is related? Example composer.json:
```json
{
    "repositories": [
        {
            "type": "composer",
            "url": "https://custom.repos.example"
        }
    ],
    "require": {
        "custom/package": "^1.0",
        "symfony/flex": "^1.0"
    }
}
```
If we run composer install or update on a directory containing just this file (and composer cache cleared), there are two problems:

- It asks for authentication for custom.repos.example, even though the credentials are already stored in a user level auth.json (e.g. without the flex package it will not ask for authentication).
- After entering authentication, the preloading does work. However, it will ask to store the credentials for the complete Url: `Do you want to store credentials for https://custom.repos.example/dist/custom/package/custom-package-v1.0.2-123456.zip in C:/Users/User/AppData/Roaming/Composer/auth.json?` rather than `custom.repos.example` (once I was even asked to store my credentials for an unrelated public github URL, but couldn't reproduce this). When answered yes, it will add a wrong entry after the existing one to the auth.json like this:
```json
{
    "http-basic": {
        "custom.repos.example": {
            "username": "user",
            "password": "pass"
        },
        "https://custom.repos.example/dist/custom/package/custom-package-v1.0.2-123456.zip": {
            "username": "user",
            "password": "pass"
        }
    }
}
```

**Comment by nicolas-grekas** [comment]
The issue is that the `$originUrl` is not computed correctly in https://github.com/symfony/flex/pull/261/files#diff-b34514797718dcfa298b2eca60fa4d15R90.

**Comment by nicolas-grekas** [comment]
Fixed by #276

**Comment by nicolas-grekas** [comment]
(please submit a separate issue, commenting on closed ones is likely to be lost.)
